modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/GiveRatingViewModel.kt

```diff
@@ -0,0 +1,111 @@
+package au.com.shiftyjelly.pocketcasts.podcasts.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import au.com.shiftyjelly.pocketcasts.models.entity.PodcastRatings
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.PodcastManager
+import au.com.shiftyjelly.pocketcasts.repositories.ratings.RatingsManager
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import javax.inject.Inject
+
+@HiltViewModel
+class GiveRatingViewModel @Inject constructor(
+    private val podcastManager: PodcastManager,
+    private val ratingsManager: RatingsManager,
+) : ViewModel() {
+
+    sealed class State {
+        object Loading : State()
+        data class Loaded(
+            val podcastUuid: String,
+            val podcastTitle: String,
+            private val _stars: Stars?,
+        ) : State() {
+
+            val stars: Stars = _stars
+                ?: Stars.TwoAndHalf // default to 2.5 stars if there is no previous rating
+
+            enum class Stars {
+                Zero,
+                Half,
+                One,
+                OneAndHalf,
+                Two,
+                TwoAndHalf,
+                Three,
+                ThreeAndHalf,
+                Four,
+                FourAndHalf,
+                Five,
+            }
+        }
+    }
+
+    private val _state = MutableStateFlow<State>(State.Loading)
+    val state = _state.asStateFlow()
+
+    fun checkIfUserCanRatePodcast(
+        podcastUuid: String,
+        onFailure: (String) -> Unit
+    ) {
+        _state.value = State.Loading
+        viewModelScope.launch(Dispatchers.IO) {
+            val newState = getPodcastInfo(podcastUuid)
```

Comment:
This currently only depends on getting a valid podcast ID (and no listening history for the podcast as I see in [this commit](https://github.com/Automattic/pocket-casts-android/commit/d45673e57acf4e9acb94b8ff774f56fb9e806bac#diff-91e13144a4addcc841353c465d672b764a7fc5003723c965066c848d3377930dR72-R73)) and this condition is subject to change, correct?

We might want to add a small test for this condition whenever it is decided.

Reply:
Good suggestion. 👍  If I don't end up removing this logic (which is what I think may be what happens on my next commit), I'll look into adding a test.